pkg/api/defaults-apiserver.go

```diff
@@ -164,6 +164,13 @@ func (cs *ContainerService) setAPIServerConfig() {
 	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.20.0-alpha.1") && o.KubernetesConfig.APIServerConfig["--service-account-issuer"] == "kubernetes.default.svc" {
 		o.KubernetesConfig.APIServerConfig["--service-account-issuer"] = "https://kubernetes.default.svc.cluster.local"
 	}
+
+	outdatedFeatureGates := []string{}
+	// Remove --feature-gate VolumeSnapshotDataSource starting with 1.22
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.22.0-alpha.1") {
+		outdatedFeatureGates = append(outdatedFeatureGates, "VolumeSnapshotDataSource")
+	}
+	removeOutdatedFeatureGates(o.KubernetesConfig.APIServerConfig, outdatedFeatureGates)
```

Comment:
This random page indicates that not only api-server should be updated: 
https://d-nix.nl/2020/06/enable-feature-gates-on-a-live-kubernetes-cluster/

I am not sure the page is correct though. Please check which components needed the flag before 1.22 and clear those as well.

Reply:
Added to all relevant components